File: day13/multiplicationTable.cpp
#include <iostream>
#include <string>
#include <sstream>

void printMultiplicationTable() {
    std::string input;
    int number;

    while (true) {
        std::cout << "Please enter a number and I will print\n"
                     "the multiplication table, to 10, for that number: ";
        std::getline(std::cin, input);
        std::cout << std::endl;

        std::stringstream ss(input);
        if (ss >> number && ss.eof()) {
            for (int i = 1; i <= 10; ++i) {
                std::cout << number << " x " << i << " = " << number * i << std::endl;
            }
            break;
        } else {
            std::cout << "\nPlease enter a number.\n\n";
        }
    }
}

int main() {
    printMultiplicationTable();
    return 0;
}

File: day13/multiplicationTable.rs
use std::io;

fn main() {
    print_multiplication_table();
}

fn print_multiplication_table() {
    loop {
        println!(); // extra line before input
        println!("Please enter a number and I will print");
        print!("the multiplication table, to 10, for that number: ");
        io::Write::flush(&mut io::stdout()).unwrap(); // flush to ensure print! shows before input

        let mut user_input = String::new();
        if io::stdin().read_line(&mut user_input).is_ok() {
            if let Ok(number) = user_input.trim().parse::<i32>() {
                println!();
                for i in 1..=10 {
                    println!("{} x {} = {}", number, i, number * i);
                }
                break;
            }
        }

        println!("\nPlease enter a number.\n");
    }
}
